src/c2goto/library/pthread_lib.c

```diff
@@ -691,3 +691,19 @@ __ESBMC_HIDE:;
   __ESBMC_atomic_end();
   return result;
 }
+
+
+int pthread_equal(pthread_t tid1, pthread_t tid2)
+
+{
+    __ESBMC_HIDE:;
+    __ESBMC_atomic_begin();
+     tid1 = __ESBMC_get_thread_id();
+     tid2 = __ESBMC_get_thread_id();
```

Comment:
Isn't this always comparing the current id with itself? This is calling [a intrinsic method that returns the current state](https://github.com/esbmc/esbmc/blob/master/src/goto-symex/builtin_functions.cpp#L470).

Also, since esbmc implementation of pthread_t is a unsigned, wouldn't using `tid1 == tid2`enought?

Reply:
Yes, it is enough. But because I used equal_ function in my code and when I run it in ESBMC I got a warning. 

**** WARNING: no body for function rand
**** WARNING: no body for function pthread_equal

since ESBMC support thread library, I tried to model the function.

Reply:
Your code should be using rand then, either change the description of this PR (that it includes the model of both rand and pthread_equal) or submit the rand model in another PR.